packages/web/src/services/competitionService.ts

```diff
@@ -0,0 +1,98 @@
+// services/competitionService.ts
+import { useState, useEffect } from "react";
+import { supabase } from "./supabaseClient";
+
+export interface Competition {
+  id: number;
+  name: string;
+  created_by: string;
+  created_at: string;
+}
+
+export function useCompetition() {
+  const [competition, setCompetition] = useState<Competition | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  const fetchCompetition = async () => {
+    try {
+      setIsLoading(true);
+      const {
+        data: { user },
+      } = await supabase.auth.getUser();
+
+      if (!user) throw new Error("No authenticated user found");
+
+      const { data, error } = await supabase
+        .from("competitions")
+        .select("*")
+        .eq("created_by", user.id)
```

Comment:
It is strange that your are returning `single` here ... a single user can create multiple competitions, no? So there might be multiple records with the same `created_by`. I think what you want is `.eq("id", id)`. You can also have the statement above but it will not be necessary once we have setup the RLS correctly.